### DISCUSSION TITLE
Installing ioncube extension

### DISCUSSION THREAD

**Discussion by franayerza** [wardenenv/warden#742] [Help & Support]
I'm trying to install ioncube in my Magento instance following this guide:
https://wetopi.com/install-ioncube-loader/

However, when I get to checking if the extension is correctly installed running php -v, I get this error in the output:

Warning: Failed loading Zend extension 'ioncube_loader_lin_8.1.so' (tried: /usr/local/lib/php/extensions/no-debug-non-zts-20210902/ioncube_loader_lin_8.1.so (Error loading shared library /usr/local/lib/php/extensions/no-debug-non-zts-20210902/ioncube_loader_lin_8.1.so: No such file or directory), /usr/local/lib/php/extensions/no-debug-non-zts-20210902/ioncube_loader_lin_8.1.so.so (Error loading shared library /usr/local/lib/php/extensions/no-debug-non-zts-20210902/ioncube_loader_lin_8.1.so.so: No such file or directory)) in Unknown on line 0

It seems there is somewhere that duplicates the .so to the file, but I'm not finding where this is happening. Everything that I've created is ok, no duplication.

**Comment by franayerza** [comment]
Just realized the error I pasted wasn't the one I meant, this is the error I'm actually dealing with:

Warning: Failed loading Zend extension 'ioncube_loader_lin_8.1.so' (tried: /usr/local/lib/php/extensions/no-debug-non-zts-20210902/ioncube_loader_lin_8.1.so (Error relocating /usr/local/lib/php/extensions/no-debug-non-zts-20210902/ioncube_loader_lin_8.1.so: unsupported relocation type 7), /usr/local/lib/php/extensions/no-debug-non-zts-20210902/ioncube_loader_lin_8.1.so.so (Error loading shared library /usr/local/lib/php/extensions/no-debug-non-zts-20210902/ioncube_loader_lin_8.1.so.so: No such file or directory)) in Unknown on line 0

**Comment by navarr** [comment]
Could you please supply your `.env` file, with any personal data redacted?

**Comment by franayerza** [comment]
Here is my .env file data:

WARDEN_ENV_NAME=iwae
WARDEN_ENV_TYPE=magento2
WARDEN_WEB_ROOT=/

TRAEFIK_DOMAIN=iwae.local
TRAEFIK_SUBDOMAIN=

WARDEN_DB=1
WARDEN_ELASTICSEARCH=1
WARDEN_OPENSEARCH=1
WARDEN_ELASTICHQ=0
WARDEN_VARNISH=1
WARDEN_RABBITMQ=1
WARDEN_REDIS=1

OPENSEARCH_VERSION=2.5
ELASTICSEARCH_VERSION=7.17
DB_DISTRIBUTION=mariadb
DB_DISTRIBUTION_VERSION=10.4
NODE_VERSION=12
COMPOSER_VERSION=2.2
PHP_VERSION=8.1
PHP_XDEBUG_3=1
RABBITMQ_VERSION=3.9
REDIS_VERSION=6.2
VARNISH_VERSION=7.1

WARDEN_SYNC_IGNORE=

WARDEN_ALLURE=0
WARDEN_SELENIUM=0
WARDEN_SELENIUM_DEBUG=0
WARDEN_BLACKFIRE=0
WARDEN_SPLIT_SALES=0
WARDEN_SPLIT_CHECKOUT=0
WARDEN_TEST_DB=0
WARDEN_MAGEPACK=0

BLACKFIRE_CLIENT_ID=
BLACKFIRE_CLIENT_TOKEN=
BLACKFIRE_SERVER_ID=
BLACKFIRE_SERVER_TOKEN=

**Comment by navarr** [comment]
Thank you.

It looks like the lack of ioncube from these images was an accidental miss.  They _should_ have ionCube automatically installed.

If you run `warden env pull` you should get a new version from today that comes with ioncube already installed.

Please let me know if that solves your problem!
